Sources/SwiftMemcache/MemcachedConnection.swift

```diff
@@ -171,19 +183,111 @@ public actor MemcachedConnection {
+    public func get<Value: MemcachedValue>(_ key: String, as valueType: Value.Type = Value.self, newTimeToLive: TimeToLive? = nil) async throws {
+        switch self.state {
+        case .initial(_, _, _, let requestContinuation, let clock),
+             .running(_, _, _, let requestContinuation, let clock):
+
+            var flags = MemcachedFlags()
+            if let newTimeToLive {
+                flags = MemcachedFlags()
+                flags.timeToLive = newTimeToLive.durationUntilExpiration(inRelationTo: clock)
+            }
+            flags.shouldReturnValue = true
+
+            let command = MemcachedRequest.GetCommand(key: key, flags: flags)
+            let request = MemcachedRequest.get(command)
+
+            _ = try await withCheckedThrowingContinuation { continuation in
+                switch requestContinuation.yield((request, continuation)) {
+                case .enqueued:
+                    break
+                case .dropped, .terminated:
+                    continuation.resume(throwing: MemcachedConnectionError.connectionShutdown)
+                default:
+                    break
+                }
+            }
+
+        case .finished:
+            throw MemcachedConnectionError.connectionShutdown
+        }
+    }
+
+    // MARK: - Fetching TTL
+
+    /// Fetch the value for a key and its TTL from the Memcache server.
+    ///
+    /// - Parameter key: The key to fetch the value and TTL for.
+    /// - Returns: A tuple containing the fetched value and its TTL, or `nil` if no value was found.
+    /// - Throws: A `MemcachedConnectionError` if the connection is shutdown.
+    public func get<Value: MemcachedValue>(_ key: String, as valueType: Value.Type = Value.self) async throws -> (Value?, TimeToLive?) {
+        switch self.state {
+        case .initial(_, _, _, let requestContinuation, let clock),
+             .running(_, _, _, let requestContinuation, let clock):
+
+            var flags = MemcachedFlags()
+            flags.shouldReturnValue = true
+            flags.shouldReturnTTL = true
+
+            let command = MemcachedRequest.GetCommand(key: key, flags: flags)
+            let request = MemcachedRequest.get(command)
+
+            var response = try await withCheckedThrowingContinuation { continuation in
+                switch requestContinuation.yield((request, continuation)) {
+                case .enqueued:
+                    break
+                case .dropped, .terminated:
+                    continuation.resume(throwing: MemcachedConnectionError.connectionShutdown)
+                default:
+                    break
+                }
+            }
+
+            let value = Value.readFromBuffer(&response.value!)
+            let ttl = response.flags?.timeToLive.map {
+                TimeToLive.expiresAt(clock.now.advanced(by: Duration.seconds($0)))
+            } ?? .indefinitely
```

Comment:
Do we get a `flags.timeToLive` back even if there is no TTL set on the key?

Reply:
Ah, I must have missed this case, as per the protocol 
`- t: return item TTL remaining in seconds (-1 for unlimited)
`
if no TTL is set it will default to -1.